Description: CA-72709: Fix logic in netmask_to_prefixlen 
 This bug caused the wrong netmask to be set for static IP configuration, in
 some cases.
Origin: upstream, https://github.com/xen-org/xen-api/commit/abed1d6321a9f56822e5c7a5ce3b183e25c90bb9
Author: Rob Hoes <rob.hoes@citrix.com>
Bug-Debian: http://bugs.debian.org/695221
Last-Update: 2013-03-23

--- xen-api-1.3.2.orig/ocaml/xapi/nm.ml
+++ xen-api-1.3.2/ocaml/xapi/nm.ml
@@ -63,7 +63,8 @@ let netmask_to_prefixlen netmask =
 			else
 				l
 		in
-		List.fold_left length 0 [a; b; c; d]
+		let masks = List.map ((-) 255) [a; b; c; d] in
+		32 - (List.fold_left length 0 masks)
 	)
 
 let determine_mtu ~__context pif_rc bridge =
